sarxarray/stack.py

```diff
@@ -109,6 +110,85 @@ def _amp_disp(self, chunk_azimuth=500, chunk_range=500):
 
         return amplitude_dispersion
 
+    def multi_look(self, window_size, method="coarsen", statistics="mean", chunk=1000, compute=True):
+        """
+        Perform multi-looking on a Stack, and return a Stack.
+
+        Parameters
+        ----------
+        window_size : tuple
+            Window size for multi-looking, in the format of (azimuth, range)
+        method : str, optional
+            Method of multi-looking, by default "coarsen"
+        statistics : str, optional
+            Statistics method for multi-looking, by default "mean"
+        chunk : int, optional
+            Chunk size in the space dimension, by default 1000
+        compute : bool, optional
+            Whether to compute the result, by default True. If False, the result
+            will be `dask.delayed.Delayed`. This is useful when the multi_look
+            is used as an intermediate result.
+
+        Returns
+        -------
+        xarray.Dataset
+            An `xarray.Dataset` with coarsen shape if `compute` is True,
+            otherwise a `dask.delayed.Delayed` object.
+        """
+        # check if window_size is valid
+        if window_size[0] > self._obj.azimuth.size or window_size[1] > self._obj.range.size:
+            warnings.warn(
+                "Window size is larger than the data size, no multi-looking is performed."
+            )
+            return self._obj
+
+        # check if window_size is smaller than chunk size
+        if window_size[0] > chunk or window_size[1] > chunk:
+            warnings.warn(
+                "Window size is larger than chunk size, no multi-looking is performed."
+            )
+            return self._obj
+
+        # add new atrrs here because Delayed objects are immutable
+        self._obj.attrs["multi-look"] = f"{method}-{statistics}"
+
+        match method:
+            case "coarsen":
+                # TODO: if boundary and size should be configurable
+                multi_looked = self._obj.coarsen(
+                    {"azimuth": window_size[0], "range": window_size[1]},
+                    boundary="trim",
+                    side="left",
+                )
+            case other:
+                raise NotImplementedError
+
+        # apply statistics
+        stat_functions = {
+            "mean": multi_looked.mean,
+            "median": multi_looked.median,
+        }
+        if statistics in stat_functions:
+            stat_function = stat_functions[statistics]
```

Comment:
Can we have the default behavior to reset the `azimuth` and `range` coordinates from 0? And keep the option to perform `stat_functions` on them?

The logic behind this is  `azimuth` and `range`  are mainly pixel coordinates, so it does not make much sense to average them. When multi-look is performed, the image size changed, then a new coordinate should be given.

Reply:
Thanks, the values are fixed using `coord_func` in `coarsen` function.